api/graphql/reservations/reservation_serializers.py

```diff
@@ -677,10 +679,33 @@ def validated_data(self):
 
     def save(self, **kwargs):
         instance = super().save(**kwargs)
+
         if instance.state in RESERVATION_STATE_EMAIL_TYPE_MAP.keys():
             send_reservation_email_task.delay(
                 instance.id, RESERVATION_STATE_EMAIL_TYPE_MAP[instance.state]
             )
+
+        if instance.state == STATE_CHOICES.REQUIRES_HANDLING:
+            recipients = get_notification_recipients(
```

Comment:
I would add this whole block to be run by a task. In `get_notification_recipients` we're looping all users and doing checks to each of them so it might take too long for the client to wait for it.

Reply:
Good idea! I created a new task for fetching recipients and sending emails 👍